### PR TITLE
Create admin dashboard

### PR DESCRIPTION
**Pull request by looneydevs**
### New features:
- Add order timer
- Add secure routes
- Add admin dashboard panel
### Amendments:
- Select options and text input suggestions rendering

### CODE REVIEW

File: src/components/MenuAuth.tsx [hunk 1]
```diff
@@ -21,24 +22,35 @@ const StyledAccountCircle = styled(AccountCircle)<any>`
...
-    console.log('succesfully signed out');
     props.history.push('/auth');
   };
 
   const handleProfileRedirect = () => {
     props.history.push('/profile');
   };
 
+  const handleAdminRedirect = () => {
+    props.history.push('/admin');
```
Comment (SobolevskyDmitry) [inline src/components/MenuAuth.tsx]: can be simplified, without {}

File: src/components/admin/AddMovieForm.tsx [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+  }, [loadedMovies, selectedMovies]);
+
+  return (
+    <AdminFormContainer title="Add Movies">
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <Fragment>
+          <SelectedDataController onSubmit={handleSubmit}>
+            <div style={{ marginBottom: '1rem' }}>
```
Comment (SobolevskyDmitry) [inline src/components/admin/AddMovieForm.tsx]: :cry:

File: src/components/admin/AddServicesForm.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  return (
+    <AdminFormContainer title="Add Services">
+      <form style={{ maxWidth: '550px' }} onSubmit={handleSubmit}>
+        <SelectField
+          id="cinema"
+          type="select"
+          options={cinemaOptions}
+          label="Choose Cinema"
+          value={cinema}
+          handleChange={(value: any) => setCinema(value)}
```
Comment (SobolevskyDmitry) [inline src/components/admin/AddServicesForm.tsx]: Can use type here

File: src/components/admin/AdminFormContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  justify-content: center;
+  padding: 1.5rem 0;
+  box-sizing: border-box;
+`;
+
+const AdminFormContainer = ({ children, title }: any) => {
+  return (
+    <Container>
+      <PageTitle text={title} />
+      <div style={{ padding: '1rem' }}>{children}</div>
```
Comment (SobolevskyDmitry) [inline src/components/admin/AdminFormContainer.tsx]: :cry:

File: src/components/admin/AdminFormContainer.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  flex: 1;
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: center;
+  padding: 1.5rem 0;
+  box-sizing: border-box;
+`;
+
+const AdminFormContainer = ({ children, title }: any) => {
+  return (
```
Comment (SobolevskyDmitry) [inline src/components/admin/AdminFormContainer.tsx]: can be simplified

File: src/components/admin/CreateSessionForm.tsx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+  const [cityTyped, setCityTyped] = useState('');
+  const [cinema, setCinema] = useState('');
+  const [hall, setHall] = useState('');
+  const [buttonDisabled, setButtonDisabled] = useState(true);
+
+  const [movieSuggestions, setMovieSuggestions] = useState(null);
+  const [citySuggestions, setCitySuggestions] = useState(null);
+  const [cinemaOptions, setCinemaOptions] = useState(null);
+  const [hallOptions, setHallOptions] = useState(null);
+  const [timeOptions, setTimeOptions] = useState(null);
```
Comment (SobolevskyDmitry) [inline src/components/admin/CreateSessionForm.tsx]: maybe too many useState ?))

File: src/pages/Admin.tsx [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+const Container = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+`;
+
+const Admin = () => {
+  const [selectedTab, setSelectedTab] = useState('add-cinema');
+
+  const getSelectedForm = (handleSnackbar: any) => {
```
Comment (looneydevs) [inline src/pages/Admin.tsx:22]: enum

**Pull request merged**